sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -844,6 +903,7 @@ public void deleteDataset(String projectId, String datasetId)
                 + "as many elements as rowList");
       }
 
+      final Set<Integer> failedIndices = new HashSet<>();
```

Comment:
How big `failedIndices` set could be? Any implication on performance impact? Probably okay since `insertAll` is only for streaming insert.

Reply:
500 elements should be fine. Thanks for the explanation.